### DISCUSSION TITLE
Front End Module on new Board (Zephyr vs. NCS)

### DISCUSSION THREAD

**Discussion by pideu-sj** [zephyrproject-rtos/zephyr#66608] [Q&A] [answered]
Hi,

I want to add a new board to zephyr, similar to the pan1783_evb/pan1783a_evb that are already available and are based on the Nordic nRF5340 chipset. This board will have a Skyworks 66407-11 front end module mounted. 

I did some configuration steps to enable the FEM support when working with ncs sdk version 2.5.0 and toolchain 2.5.0. With these changes to the project, I got the sample running with the front end module.

Now I want to provide this knowledge inside a board definition to the customers. But when adding the following needed configurations to the defconfig of the cpunet core, 

```
CONFIG_MPSL=y
CONFIG_MPSL_FEM=y
CONFIG_MPSL_FEM_SIMPLE_GPIO=y
```

it would no longer be possible to build something like a simple blinky on zephyr sdk 0.16.3 (v3.5.99), because these symbols are undefined in zephyr.

How can I provide this information to the board definition in zephyr, so they are available later in the newest ncs release, but at the same time zephyr is ok with that?

Maybe some Nordic experts like, @nordicjm or @anangl can help here?
 
Thanks a lot!

**Comment by nordicjm** [comment, marked as answer]
You can set it up in devicetree as per the details on this page: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/net/wireless/nordic%2Cnrf-radio.yaml which would then allow you to submit the board to upstream zephyr. It would then be usable in ncs when an upmerge happens and pulls those changes in, as to if you would need to set different Kconfig or not I am not sure.

Alternatively you could create your board specifically for ncs and store it in your own `boards` repo then create a west manifest that gets ncs and your boards repo which allows your customers to use it, and would just tell them the manifest repository to use.

**Comment by pideu-sj** [comment]
Ah, okay, got it. 
After some testing I noticed, that the mentioned Kconfig's
```
CONFIG_MPSL=y
CONFIG_MPSL_FEM=y
CONFIG_MPSL_FEM_SIMPLE_GPIO=y
```
don't need to be set for ncs. It looks like the KConfig in `v2.5.0/nrf/subsys/mpsl/fem/Kconfig` notices when a generic-fem-two-ctrl-pins node is available in the devicetree and enables the needed configs by itself, which is great. So I can simply just add the fem node to the devicetrees in zephyr and just do the PR.

Thanks a lot!
